src/OrchardCore.Modules/OrchardCore.ContentTypes/Services/ContentDefinitionService.cs

```diff
@@ -337,7 +337,7 @@ public void AlterPartFieldsOrder(ContentPartDefinition partDefinition, string[]
                     var fieldDefinition = partDefinition.Fields.FirstOrDefault(x => x.Name == fieldNames[i]);
                     type.WithField(fieldNames[i], field =>
                     {
-                        field.WithSetting("Position", i.ToString());
+                        field.MergeSettings<ContentPartFieldSettings>(x => x.Position = i.ToString());
                     });
```

Comment:
Position should stay as string, because `before` is a valid position (although not common!)